include/xsimd/stl/algorithms.hpp

```diff
@@ -11,125 +11,187 @@
 #ifndef XSIMD_ALGORITHMS_HPP
 #define XSIMD_ALGORITHMS_HPP
 
+#include <type_traits>
 #include "../memory/xsimd_load_store.hpp"
 
 namespace xsimd
 {
-    template <class I1, class I2, class O1, class UF>
-    void transform(I1 first, I2 last, O1 out_first, UF&& f)
-    {
-        using value_type = typename std::decay<decltype(*first)>::type;
-        using traits = simd_traits<value_type>;
-        using batch_type = typename traits::type;
+    template <typename... >
+    using void_t = void;
 
-        std::size_t size = static_cast<std::size_t>(std::distance(first, last));
-        std::size_t simd_size = traits::size;
+    template <class T, class = void>
+    struct has_increment : std::false_type {};
 
-        const auto* ptr_begin = &(*first);
-        auto* ptr_out = &(*out_first);
+    template <class T>
+    struct has_increment<T, void_t<decltype(++std::declval<T>())>> : std::true_type {};
 
-        std::size_t align_begin = xsimd::get_alignment_offset(ptr_begin, size, simd_size);
-        std::size_t out_align = xsimd::get_alignment_offset(ptr_out, size, simd_size);
-        std::size_t align_end = align_begin + ((size - align_begin) & ~(simd_size - 1));
+    template <class T>
+    using enable_if_increment = typename std::enable_if<has_increment<T>::value>::type;
+
+    template <class T>
+    using enable_if_not_increment = typename std::enable_if<!has_increment<T>::value>::type;
```

Comment:
I think it would be better to move these metafunctions in some `detail` namespace, they're not supposed to be part of the API.